types/is-file-esm/index.d.ts

```diff
@@ -0,0 +1,27 @@
+// Type definitions for is-file-esm 1.0
+// Project: https://github.com/davidmarkclements/is-file-esm
+// Definitions by: Leon Si <https://github.com/leonzalion>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+interface Result {
+    esm: boolean;
+    type?: 'module' | 'commonjs';
```

Comment:
it's rather explicit as using switch, based on language feature (Destructuring assignment):
```suggestion
    type: 'module' | 'commonjs' | undefined;
```
https://github.com/davidmarkclements/is-file-esm/blob/83ec13a445422820f4047801764a26f247293411/index.js#L70
e.g.:
```js
const { type } = {};
console.log({esm: false, type })
> {esm: false, type: undefined}
```
(same for array destruct).